r/src/filesystem.cpp

```diff
@@ -33,6 +33,12 @@ const char* r6_class_name<fs::FileSystem>::get(
     return "LocalFileSystem";
   } else if (type_name == "s3") {
     return "S3FileSystem";
+  } else if (type_name == "gcs") {
+    return "GcsFileSystem";
+  } else if (type_name == "abfs") {
+    return "AzureBlobFileSystem";
+  } else if (type_name == "hdfs") {
+    return "HadoopFileSystem";
```

Comment:
Put them here for completeness but we should probably comment this block out for now since these R6 classes don't exist yet and the default fallback to FileSystem might allow things to Just Work if you're linking to a C++ library that has support for these built.